packages/store/src/Ability.type.ts

```diff
@@ -0,0 +1,18 @@
+export type Ability<T> =
+  | ({
+      enabled: true;
+    } & T)
+  | {
+      enabled: false;
+      userMessage: string;
```

Comment:
A message to show in UI when this ability is not enabled.

For example, `"Webcam permissions have not been granted"` or `"Store was not configured with any providers that support uploads."`